src/libexpr/primops.cc

```diff
@@ -2029,7 +2029,14 @@ static void prim_readDir(EvalState & state, const PosIdx pos, Value * * args, Va
     // Retrieve directory entries for all nodes in a directory.
     // This is similar to `getFileType` but is optimized to reduce system calls
     // on many systems.
-    auto entries = path.readDirectory();
+    SourceAccessor::DirEntries entries;
+    try {
+        entries = path.readDirectory();
+    } catch (std::filesystem::filesystem_error & e) {
+        // std::filesystem::filesystem_error already prints the directory path
+        state.error<EvalError>("reading directory content failed: %1%", e.what())
```

Comment:
```suggestion
        state.error<EvalError>("reading directory content failed:\n%1%", e.what())
```

Given the messages can get quite long maybe we should break lines at some point?

Reply:
`std::filesystem` produces an overly detailed message that just bad. Users don't care about iterators, and those four colons and brackets aren't Nix-like.
A newline does nothing to improve that.
We'll have to catch common exceptions and format them properly. Good news: most of these are `ENOTFOUND` anyway, so it's easy to make a difference.
Doesn't have to be done in this PR, but it makes the newline irrelevant in my view.

Reply:
Check the latest version. This is no longer the case. Error messages are nicer again.